sdks/python/apache_beam/transforms/window.py

```diff
@@ -369,7 +369,7 @@ def merge(self, merge_context):
     to_merge = []
     for w in sorted(merge_context.windows, key=lambda w: w.start):
       if to_merge:
-        if end > w.start:
+        if end > w.start:  # pylint: disable=used-before-assignment
```

Comment:
Here, explicitly assigning `end` to some idempotent element would improve readability and remove the need for the pylint suppression. Am I missing something special?